redhat-appstudio/remote-secret
# Target secret overrides

### What does this PR do?
This adds the ability override the secret spec per-target.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-605

### How to test this PR?
1. Create a RS like this:
   ```yaml
   apiVersion: appstudio.redhat.com/v1beta1
   kind: RemoteSecret
   metadata:
     name: target-override-test
     namespace: default
   spec:
     secret:
       generateName: to-test-
     targets:
     - namespace: default
     - namespace: default
       secret:
         name: with-to
         labels:
           test: "yes"
   stringData:
     k1: "v1"
     k2: "v2"
   ```
2. Make sure that 2 secrets exist in the default namespace: one called `with-to` and another with the name prefixed with `to-test-`. The `with-to` secret should have a label called `test` with value `yes`.
3. The override on the target can change the `generateName`, `name` (which takes precedence over the generateName defined either in spec or in the override (as is the case with ordinary k8s objects, too)), `labels`, `annotations` and also `linkedTo`. Modify the example above with the overrides to those to test them, too.

Test image: `quay.io/lkrejci/remote-secret-controller:svpi-605`
